### PR TITLE
Components update

### PR DESCRIPTION
**Pull request by bedakb**
Mostly visual updates.Affected components are:
 - Checkbox
 - Multi Level Checkbox Editor
 - Two Box Multiselect

### CODE REVIEW

File: src/checkbox-table/checkbox-table.tsx [hunk 1]
```diff
@@ -34,7 +33,8 @@ interface Props {
     onChange: (data: CheckboxTableData) => void;
     selectAllLabel: string;
     isDisabledCell?: (columnKey: string, rowKey: string) => boolean;
-    selectAllColumnLabel: string;
+    /** @deprecated */
+    selectAllColumnLabel?: string;
```
Comment (BenjaminEckardt) [inline src/checkbox-table/checkbox-table.tsx]: Since we already have some breaking changes (even though it's just regarding styles) i'd prefer to remove it.

File: src/checkbox/checkbox.tsx [hunk 2]
```diff
@@ -14,16 +15,19 @@ const Checkbox: React.FunctionComponent<Props> = (props) => {
...
+                id={props.id}
+                type="checkbox"
+                checked={props.value}
+                data-qa={`checkbox-${props.id}`}
+                onChange={changeHandler}
+                disabled={props.disabled}
+                className={`checkbox ${props.className}`}
+                aria-checked={props.value}
+            />
+            <label htmlFor={props.id}>{props.label || ''}</label>
```
Comment (BenjaminEckardt) [inline src/checkbox/checkbox.tsx]: Introducing the label here seems a bit weird. I'd expect these two things to be separate. Anything speaking against it? Are we already doing it somewhere like that?

File: src/form/form.less [hunk 3]
```diff
@@ -10,7 +10,8 @@
 
     input[type='text'], textarea {
         &::placeholder {
-            color: @gray-mid;
+            color: @gray-darker;
+            opacity: 1;
```
Comment (BenjaminEckardt) [inline src/form/form.less]: Why is that needed?

File: src/multi-level-checkbox-editor/multi-level-checkbox-editor.less [hunk 4]
```diff
@@ -25,20 +26,26 @@
...
+            padding: 10px 15px;
             border: @default-border;
+            border-bottom: 0;
+            background: @white;
 
             .header-label {
                 margin: 0;
             }
+
+            label {
```
Comment (BenjaminEckardt) [inline src/multi-level-checkbox-editor/multi-level-checkbox-editor.less]: I'd prefer a class

File: src/multi-level-checkbox-editor/multi-level-checkbox-editor.tsx [hunk 5]
```diff
@@ -11,7 +11,8 @@ interface Props {
     id: string;
     nodes: Array<Node>;
     onNodesChange: (nodes: Array<Node>) => void;
-    selectAllLabel: string;
+    /** @deprecated */
+    selectAllLabel?: string;
```
Comment (BenjaminEckardt) [inline src/multi-level-checkbox-editor/multi-level-checkbox-editor.tsx]: Same a above: just delete.

File: src/checkbox/checkbox.tsx (hunk 2 above)
Comment (BenjaminEckardt) [reply src/checkbox/checkbox.tsx]: Also not tested, currently.

File: src/multi-level-checkbox-editor/node-box.tsx [hunk 6]
```diff
@@ -11,7 +10,8 @@ interface Props {
     onSelectAll: (node: Node, value: boolean) => void;
     onNodeClick: (node: Node) => void;
     setNodeValue: (node: Node, value: boolean) => void;
-    selectAllLabel: string;
+    /** @deprecated */
+    selectAllLabel?: string;
```
Comment (BenjaminEckardt) [inline src/multi-level-checkbox-editor/node-box.tsx]: same

File: src/two-box-multiselect/two-box-multiselect.less [hunk 7]
```diff
@@ -18,28 +18,31 @@
...
             margin-bottom: 10px;
         }
 
         .two-box-multiselect-select-box {
             border: @default-border;
             background: @white;
+            box-shadow: 0 1px 1px rgba(0, 0, 0, 0.05);
 
             .two-box-multiselect-filter-bar {
                 @border-gap: 10px;
```
Comment (BenjaminEckardt) [inline src/two-box-multiselect/two-box-multiselect.less]: Seems unused

File: src/checkbox/checkbox.tsx (hunk 2 above)
Comment (bedakb) [reply src/checkbox/checkbox.tsx]: It's used mainly for creating custom styled checkbox component.

File: src/form/form.less (hunk 3 above)
Comment (bedakb) [reply src/form/form.less]: Change in design :)

Comment (bedakb) [reply src/form/form.less]: ops, I have no idea why I put It there.

File: src/multi-level-checkbox-editor/multi-level-checkbox-editor.less (hunk 4 above)
Comment (bedakb) [reply src/multi-level-checkbox-editor/multi-level-checkbox-editor.less]: That would mean that we need to either allow Checkbox component to accept className as prop and then show It on label, or declare class directly on label in checkbox component.
Are you ok with one of those options ?

File: src/checkbox/checkbox.tsx (hunk 2 above)
Comment (bedakb) [reply src/checkbox/checkbox.tsx]: Clarified.
